saved tracks

Bachhofer opened the pull request:
added feature to analyze saved tracks and also fixed some bugs i found on the way.
This is still work in progress but I would appreciate some feedback on this.

NLeRoy917 commented on lib/Spotify.py:
```diff
@@ -20,8 +20,33 @@ def get_playlists(self):
+
+            playlist["collaborative"] = False
+            playlist["description"] = "Saved tracks"
+            playlist["name"] = "Saved tracks"
```
So so nit-picky but just a change from "Saved tracks" to "Saved Tracks" (Capitalize tracks)

NLeRoy917 reviewed (approved):
Awesome! Thank you so much. I have been thinking about this on and off but wasn't sure how to implement it really. I like the idea of adding an endpoint for playlists - was probably needed anyway and clever to treat the saved tracks like a playlist.

Are you still working on this - should I wait to merge?

Bachhofer commented:
The conversion from saved tracks to playlist still needs a bit of work and I haven't tested yet if everything works properly. So don't merge this yet.

NLeRoy917 commented:
Sweet. I'll hold off - I pulled the changes and checked it out and it worked on my end. Let me know if you need help/make progress.

Bachhofer commented:
The conversion from saved tracks to a playlist object is now complete. I also cleaned up my code a bit and implemented your requested changes. So I you can merge this if everything is fine.

During testing I did come across an issue. When the playlist has only a few songs the lyrics fetching does not work when the page is reloaded in quick succession.  This is related to [this issue](NTag/lyrics.ovh#10). There's nothing we can do about this except switching to a different lyrics api. I think this issue can be ignored for now.

NLeRoy917 reviewed (approved):
This looks great. Thank you so much! Regarding that issue, I'm fine to ignore it for now ... that API is pretty hacky if I remember correctly, but It is faster than any other solution I could find and it isn't rate limited I believe... I can look into it more and see if I can't figure out why it's returning empty lyircs.

NLeRoy917 commented:
I'm going to go ahead and merge and deploy this

NLeRoy917 commented:
Deployed. I tested it on my Mac and it worked, but I also tested it on a windows machine I have and it was throwing a 500 error for insufficient client scope when pulling playlists. Worked on my mobile too.

I cleared the cache and cookies on the Windows machine and then it worked again - so somewhere the scope was being stored and it needed to be cleared to work again... Just an FYI
